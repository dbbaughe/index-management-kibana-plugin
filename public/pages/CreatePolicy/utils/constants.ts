/*
 * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

export const DEFAULT_POLICY = JSON.stringify(
  {
    policy: {
      description: "Default policy",
      default_state: "Ingest",
      states: [
        {
          name: "Ingest",
          actions: [{ rollover: { min_doc_count: 10000000 } }],
          transitions: [{ state_name: "Search" }],
        },
        {
          name: "Search",
          actions: [],
          transitions: [
            {
              state_name: "Delete",
              conditions: { min_index_age: "30d" },
            },
          ],
        },
        {
          name: "Delete",
          actions: [{ delete: {} }],
          transitions: [],
        },
      ],
    },
  },
  null,
  4
);
