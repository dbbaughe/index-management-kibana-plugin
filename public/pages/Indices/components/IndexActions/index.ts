import IndexActions from './IndexActions';

export default IndexActions;
