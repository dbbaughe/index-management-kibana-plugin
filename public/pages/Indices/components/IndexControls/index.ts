import IndexControls from './IndexControls';

export default IndexControls;
