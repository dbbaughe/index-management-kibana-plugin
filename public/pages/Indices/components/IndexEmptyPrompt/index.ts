import IndexEmptyPrompt from './IndexEmptyPrompt';

export default IndexEmptyPrompt;
