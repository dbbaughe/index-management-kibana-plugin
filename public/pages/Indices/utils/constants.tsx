/*
 * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from "react";
import { EuiHealth } from "@elastic/eui";

export const DEFAULT_PAGE_SIZE_OPTIONS = [5, 10, 20, 50];

export const DEFAULT_QUERY_PARAMS = {
  from: 0,
  size: 20,
  search: "",
  sortField: "name",
  sortDirection: "desc",
};

const HEALTH_TO_COLOR: {
  [health: string]: string;
  green: string;
  yellow: string;
  red: string;
  undefined: string;
} = {
  green: "success",
  yellow: "warning",
  red: "danger",
  undefined: "subdued",
};

export const indicesColumns = [
  {
    field: "index",
    name: "Index",
    sortable: true,
    truncateText: true,
    textOnly: true,
    width: "250px",
  },
  {
    field: "health",
    name: "Health",
    sortable: true,
    truncateText: true,
    textOnly: true,
    align: "right",
    render: (health: string) => <EuiHealth color={HEALTH_TO_COLOR[health]}>{health}</EuiHealth>,
  },
  {
    field: "status",
    name: "Status",
    sortable: true,
    truncateText: true,
    textOnly: true,
    align: "right",
  },
  {
    field: "store.size",
    name: "Total size",
    sortable: true,
    truncateText: true,
    textOnly: true,
    dataType: "number",
  },
  {
    field: "pri.store.size",
    name: "Primaries size",
    sortable: true,
    truncateText: true,
    textOnly: true,
    dataType: "number",
  },
  {
    field: "docs.count",
    name: "Total documents",
    sortable: true,
    truncateText: true,
    textOnly: true,
    dataType: "number",
  },
  {
    field: "docs.deleted",
    name: "Deleted documents",
    sortable: true,
    truncateText: true,
    textOnly: true,
    dataType: "number",
  },
  {
    field: "pri",
    name: "Primaries",
    sortable: true,
    truncateText: true,
    textOnly: true,
    dataType: "number",
  },
  {
    field: "rep",
    name: "Replicas",
    sortable: true,
    truncateText: true,
    textOnly: true,
    dataType: "number",
  },
];
