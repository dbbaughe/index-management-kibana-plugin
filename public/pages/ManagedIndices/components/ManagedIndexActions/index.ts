import ManagedIndexActions from './ManagedIndexActions';

export default ManagedIndexActions;
