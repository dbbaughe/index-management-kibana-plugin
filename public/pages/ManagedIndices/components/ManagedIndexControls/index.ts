import ManagedIndexControls from './ManagedIndexControls';

export default ManagedIndexControls;
