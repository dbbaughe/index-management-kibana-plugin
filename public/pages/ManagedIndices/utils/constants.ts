export const DEFAULT_PAGE_SIZE_OPTIONS = [5, 10, 20, 50];

export const DEFAULT_QUERY_PARAMS = {
  from: 0,
  size: 20,
  search: '',
  sortField: 'name',
  sortDirection: 'desc',
};

export const ACTIONS: {
  [action: string]: string;
  rollover: string;
  delete: string;
  open: string;
  close: string;
  transition: string;
} = {
  rollover: 'Rollover',
  delete: 'Delete',
  transition: 'Transition',
  open: 'Open',
  close: 'Close',
};
