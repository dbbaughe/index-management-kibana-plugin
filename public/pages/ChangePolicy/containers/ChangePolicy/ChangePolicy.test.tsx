/*
 * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, wait } from "@testing-library/react";
import chrome from "ui/chrome";
import { HashRouter as Router, Redirect, Route, Switch } from "react-router-dom";
import ChangePolicy from "./ChangePolicy";
import { browserServicesMock } from "../../../../../test/mocks";
import { BREADCRUMBS, ROUTES } from "../../../../utils/constants";
import { ServicesConsumer, ServicesContext } from "../../../../services";
import { ModalProvider, ModalRoot } from "../../../../components/Modal";

function renderWithRouter(Component: React.ComponentType<any>) {
  return {
    ...render(
      <Router>
        <Switch>
          <Route
            path={ROUTES.CHANGE_POLICY}
            render={props => (
              <ServicesContext.Provider value={browserServicesMock}>
                <ModalProvider>
                  <ServicesConsumer>{services => services && <ModalRoot services={services} />}</ServicesConsumer>
                  <ServicesConsumer>
                    {({ managedIndexService, indexService }: any) => (
                      <Component indexService={indexService} managedIndexService={managedIndexService} {...props} />
                    )}
                  </ServicesConsumer>
                </ModalProvider>
              </ServicesContext.Provider>
            )}
          />
          <Redirect from="/" to={ROUTES.CHANGE_POLICY} />
        </Switch>
      </Router>
    ),
  };
}

describe("<ChangePolicy /> spec", () => {
  it("renders the component", async () => {
    const { container } = renderWithRouter(ChangePolicy);

    await wait();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("sets breadcrumbs when mounting", async () => {
    renderWithRouter(ChangePolicy);

    expect(chrome.breadcrumbs.set).toHaveBeenCalledTimes(1);
    expect(chrome.breadcrumbs.set).toHaveBeenCalledWith([
      BREADCRUMBS.INDEX_MANAGEMENT,
      BREADCRUMBS.MANAGED_INDICES,
      BREADCRUMBS.CHANGE_POLICY,
    ]);
  });
});
