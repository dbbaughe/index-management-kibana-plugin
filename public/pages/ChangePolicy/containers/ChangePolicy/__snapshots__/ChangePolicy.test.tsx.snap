// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChangePolicy /> spec renders the component 1`] = `
<div
  style="padding: 0px 25px;"
>
  <h1
    class="euiTitle euiTitle--large"
  >
    Change policy
  </h1>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiPanel euiPanel--paddingMedium"
    style="padding-left: 0px; padding-right: 0px;"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
      style="padding: 0px 10px;"
    >
      <div
        class="euiFlexItem"
      >
        <h3
          class="euiTitle euiTitle--small"
        >
          Managed indices
        </h3>
      </div>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
    />
    <div
      style="padding: initial;"
    >
      <div
        style="padding-left: 10px;"
      >
        <div
          class="euiText euiText--extraSmall"
        >
          <p>
            When a managed index changes policy, the new policy will take affect after the managed index is finished running the current state or the new policy has the same structure as the current one.
          </p>
        </div>
        <div
          class="euiSpacer euiSpacer--m"
        />
        <div
          class="euiFormRow"
          id="some_make_id-row"
        >
          <label
            aria-invalid="false"
            class="euiFormLabel"
            for="some_make_id"
          >
            Managed indices
          </label>
          <div
            aria-describedby="some_make_id-help"
            aria-expanded="false"
            aria-haspopup="listbox"
            class="euiComboBox"
            role="combobox"
          >
            <div
              class="euiFormControlLayout"
            >
              <div
                class="euiFormControlLayout__childrenWrapper"
              >
                <div
                  class="euiComboBox__inputWrap euiComboBox__inputWrap-isClearable"
                  data-test-subj="comboBoxInput"
                  tabindex="-1"
                >
                  <p
                    class="euiComboBoxPlaceholder"
                  >
                    Choose managed indices
                  </p>
                  <div
                    class="euiComboBox__input"
                    style="font-size: 14px; display: inline-block;"
                  >
                    <input
                      data-test-subj="comboBoxSearchInput"
                      id="some_make_id"
                      role="textbox"
                      style="box-sizing: content-box; width: 2px;"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
                <div
                  class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
                >
                  <button
                    aria-label="Open list of options"
                    class="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                    data-test-subj="comboBoxToggleListButton"
                  >
                    <svg
                      aria-hidden="true"
                      class="euiIcon euiIcon--medium euiFormControlLayoutCustomIcon__icon"
                      focusable="false"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                      xmlns:xlink="http://www.w3.org/1999/xlink"
                    >
                      <defs>
                        <path
                          d="M13.069 5.157L8.384 9.768a.546.546 0 0 1-.768 0L2.93 5.158a.552.552 0 0 0-.771 0 .53.53 0 0 0 0 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 0 0 0-.76.552.552 0 0 0-.771 0z"
                          id="arrow_down-a"
                        />
                      </defs>
                      <use
                        fill-rule="nonzero"
                        xlink:href="#arrow_down-a"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="euiFormHelpText euiFormRow__text"
            id="some_make_id-help"
          >
            You can use a * as a wildcard in your index pattern
          </div>
        </div>
        <div
          class="euiFormRow"
          id="some_make_id-row"
        >
          <label
            class="euiFormLabel"
            for="some_make_id"
          >
            State filters
          </label>
          <div
            aria-describedby="some_make_id-help"
            aria-expanded="false"
            aria-haspopup="listbox"
            class="euiComboBox"
            role="combobox"
          >
            <div
              class="euiFormControlLayout"
            >
              <div
                class="euiFormControlLayout__childrenWrapper"
              >
                <div
                  class="euiComboBox__inputWrap euiComboBox__inputWrap-isClearable"
                  data-test-subj="comboBoxInput"
                  tabindex="-1"
                >
                  <p
                    class="euiComboBoxPlaceholder"
                  >
                    Choose state filters
                  </p>
                  <div
                    class="euiComboBox__input"
                    style="font-size: 14px; display: inline-block;"
                  >
                    <input
                      data-test-subj="comboBoxSearchInput"
                      id="some_make_id"
                      role="textbox"
                      style="box-sizing: content-box; width: 2px;"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
                <div
                  class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
                >
                  <button
                    aria-label="Open list of options"
                    class="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                    data-test-subj="comboBoxToggleListButton"
                  >
                    <svg
                      aria-hidden="true"
                      class="euiIcon euiIcon--medium euiFormControlLayoutCustomIcon__icon"
                      focusable="false"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                      xmlns:xlink="http://www.w3.org/1999/xlink"
                    >
                      <defs>
                        <path
                          d="M13.069 5.157L8.384 9.768a.546.546 0 0 1-.768 0L2.93 5.158a.552.552 0 0 0-.771 0 .53.53 0 0 0 0 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 0 0 0-.76.552.552 0 0 0-.771 0z"
                          id="arrow_down-a"
                        />
                      </defs>
                      <use
                        fill-rule="nonzero"
                        xlink:href="#arrow_down-a"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="euiFormHelpText euiFormRow__text"
            id="some_make_id-help"
          >
            Apply new policy only on managed indices in these states.
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiPanel euiPanel--paddingMedium"
    style="padding-left: 0px; padding-right: 0px;"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
      style="padding: 0px 10px;"
    >
      <div
        class="euiFlexItem"
      >
        <h3
          class="euiTitle euiTitle--small"
        >
          New policy
        </h3>
      </div>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
    />
    <div
      style="padding: initial;"
    >
      <div
        style="padding-left: 10px;"
      >
        <div
          class="euiText euiText--extraSmall"
        >
          <p>
            Select a new policy. Choose between switching to a new state after changing policies or staying in the current state.
          </p>
        </div>
        <div
          class="euiSpacer euiSpacer--s"
        />
        <div
          class="euiFormRow"
          id="some_make_id-row"
        >
          <label
            aria-invalid="false"
            class="euiFormLabel"
            for="some_make_id"
          >
            Policy
          </label>
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            class="euiComboBox"
            role="combobox"
          >
            <div
              class="euiFormControlLayout"
            >
              <div
                class="euiFormControlLayout__childrenWrapper"
              >
                <div
                  class="euiComboBox__inputWrap euiComboBox__inputWrap--noWrap euiComboBox__inputWrap-isClearable"
                  data-test-subj="comboBoxInput"
                  tabindex="-1"
                >
                  <p
                    class="euiComboBoxPlaceholder"
                  >
                    Choose a policy
                  </p>
                  <div
                    class="euiComboBox__input"
                    style="font-size: 14px; display: inline-block;"
                  >
                    <input
                      data-test-subj="comboBoxSearchInput"
                      id="some_make_id"
                      role="textbox"
                      style="box-sizing: content-box; width: 2px;"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
                <div
                  class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
                >
                  <button
                    aria-label="Open list of options"
                    class="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                    data-test-subj="comboBoxToggleListButton"
                  >
                    <svg
                      aria-hidden="true"
                      class="euiIcon euiIcon--medium euiFormControlLayoutCustomIcon__icon"
                      focusable="false"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                      xmlns:xlink="http://www.w3.org/1999/xlink"
                    >
                      <defs>
                        <path
                          d="M13.069 5.157L8.384 9.768a.546.546 0 0 1-.768 0L2.93 5.158a.552.552 0 0 0-.771 0 .53.53 0 0 0 0 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 0 0 0-.76.552.552 0 0 0-.771 0z"
                          id="arrow_down-a"
                        />
                      </defs>
                      <use
                        fill-rule="nonzero"
                        xlink:href="#arrow_down-a"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="euiRadio euiRadioGroup__item"
          >
            <input
              checked=""
              class="euiRadio__input"
              id="current"
              type="radio"
              value=""
            />
            <div
              class="euiRadio__circle"
            />
            <label
              class="euiRadio__label"
              for="current"
            >
              Continue from current state
            </label>
          </div>
          <div
            class="euiRadio euiRadioGroup__item"
          >
            <input
              class="euiRadio__input"
              disabled=""
              id="state"
              type="radio"
              value=""
            />
            <div
              class="euiRadio__circle"
            />
            <label
              class="euiRadio__label"
              for="state"
            >
              Start from a chosen state after changing policies
            </label>
          </div>
        </div>
        <div
          class="euiSpacer euiSpacer--s"
        />
        <div
          class="euiFormRow"
          id="some_make_id-row"
        >
          <label
            class="euiFormLabel"
            for="some_make_id"
          >
            Start state
          </label>
          <div
            class="euiFormControlLayout"
          >
            <div
              class="euiFormControlLayout__childrenWrapper"
            >
              <select
                aria-describedby="some_make_id-help"
                aria-label="Start state for new policy"
                class="euiSelect"
                disabled=""
                id="some_make_id"
              />
              <div
                class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
              >
                <span
                  class="euiFormControlLayoutCustomIcon"
                >
                  <svg
                    aria-hidden="true"
                    class="euiIcon euiIcon--medium euiFormControlLayoutCustomIcon__icon"
                    focusable="false"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                    xmlns:xlink="http://www.w3.org/1999/xlink"
                  >
                    <defs>
                      <path
                        d="M13.069 5.157L8.384 9.768a.546.546 0 0 1-.768 0L2.93 5.158a.552.552 0 0 0-.771 0 .53.53 0 0 0 0 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 0 0 0-.76.552.552 0 0 0-.771 0z"
                        id="arrow_down-a"
                      />
                    </defs>
                    <use
                      fill-rule="nonzero"
                      xlink:href="#arrow_down-a"
                    />
                  </svg>
                </span>
              </div>
            </div>
          </div>
          <div
            class="euiFormHelpText euiFormRow__text"
            id="some_make_id-help"
          >
            Select a state to start from after changing policies
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
  >
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <button
        class="euiButton euiButton--primary euiButton--fill"
        data-test-subj="changePolicyChangeButton"
        type="button"
      >
        <span
          class="euiButton__content"
        >
          <span
            class="euiButton__text"
          >
            Change
          </span>
        </span>
      </button>
    </div>
  </div>
</div>
`;
